grails-app/controllers/org/pih/warehouse/outbound/PackingListController.groovy

```diff
@@ -0,0 +1,22 @@
+package org.pih.warehouse.outbound
+
+import grails.converters.JSON
+import org.pih.warehouse.core.UploadService
+import org.pih.warehouse.importer.DataImporter
+import org.pih.warehouse.importer.ImportDataCommand
+import org.pih.warehouse.importer.PackingListExcelImporter
+import org.springframework.web.multipart.MultipartFile
+
+class PackingListController {
+
+    UploadService uploadService
+
+    def importPackingList(ImportDataCommand command) {
+        MultipartFile importFile = command.importFile
+        File localFile = uploadService.createLocalFile(importFile.originalFilename)
+        importFile.transferTo(localFile)
+        DataImporter packingListImporter = new PackingListExcelImporter(localFile.absolutePath)
```

Comment:
One other thing we can do to improve the code would be to get rid of the file handling above and just use an input stream. This avoids the "bit of mess" and also starts us down the path of removing the file system as a dependency 

Tech Debt: The application shouldn't write to the local file system.

The way to do that would be to create a new constructor (or replace the existing one)
```
PackingListExcelImporter(InputStream inputStream) {
        super()
        read(inputStream)
        excelImportService = Holders.grailsApplication.mainContext.getBean("excelImportService")
        categoryImportDataService = Holders.grailsApplication.mainContext.getBean("categoryImportDataService")
    }
```
and then reduce the code above to two lines, maybe with some validation of the command for an empty file (and other obvious error handling).
```
DataImporter dataImporter = new PackingListExcelImporter(command.importFile.inputStream)
render([data: dataImporter.data] as JSON)
```



As far as I recall, we didn't have that available to us in Grails 1.3.9 but the read is there in the Grails 3 version of the plugin.